web/third/bootstrap/js/bootstrap.js

```diff
@@ -393,10 +393,13 @@
 
       isActive = $parent.hasClass('open')
 
-      if (!isActive || (isActive && e.keyCode == 27)) {
-        if (e.which == 27) $parent.find(toggle).focus()
+      if (isActive && e.keyCode == 27) {
         return $this.click()
       }
+      else if(!isActive) {
+        this.blur()
+        return
+      }
 
       $items = $('[role=menu] li:not(.divider):visible a', $parent)
```

Comment:
We don't want to modify this file to fix the issue, since this is third party code. You should be able to fix this in hotkeys alone as I see it.

Reply:
Info: I keydown event is not registering in the ```hotkey.js``` file. i.e when esc button is clicked, ```bootstrap.js``` is triggered and the event does not even enter the ```hotkey.js``` file.
Will try to fix it.